app/models/schema_contract/validator.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module SchemaContract
+  class Validator
+    class SchemaContractValidationError < StandardError; end
+
+    def initialize(record_id)
+      @record_id = record_id
+    end
+
+    def validate
+      errors = JSON::Validator.fully_validate(parsed_schema, record.response)
+      if errors.any?
+        @result = :schema_errors_found
+        record.update(error_details: errors)
+        detailed_message = { error_type: 'Schema discrepancy found', record_id: @record_id, response: record.response,
+                             details: errors }
+        raise SchemaContractValidationError, detailed_message
+      else
+        @result = :success
+      end
+    rescue SchemaContractValidationError => e
```

Comment:
I don't love the rescue and re-raise, but I wanted to handle specific cases and also rescue any unexpected errors below.